File: Basic Coding Problems/01 Fundamentals/018 Write a program to count the total number of notes in a given amount/Java/count_notes.java
import java.util.Scanner;

public class count_notes {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the amount: ");
        int amount = scanner.nextInt();

        int notesOf500 = amount / 500;
        amount %= 500;

        int notesOf200 = amount / 200;
        amount %= 200;

        int notesOf100 = amount / 100;
        amount %= 100;

        int notesOf50 = amount / 50;
        amount %= 50;

        int notesOf20 = amount / 20;
        amount %= 20;

        int notesOf10 = amount / 10;
        amount %= 10;

        int notesOf5 = amount / 5;
        amount %= 5;

        int notesOf2 = amount / 2;
        amount %= 2;

        int notesOf1 = amount;

        System.out.println("The number of 500 notes is: " + notesOf500);
        System.out.println("The number of 200 notes is: " + notesOf200);
        System.out.println("The number of 100 notes is: " + notesOf100);
        System.out.println("The number of 50 notes is: " + notesOf50);
        System.out.println("The number of 20 notes is: " + notesOf20);
        System.out.println("The number of 10 notes is: " + notesOf10);
        System.out.println("The number of 5 notes is: " + notesOf5);
        System.out.println("The number of 2 notes is: " + notesOf2);
        System.out.println("The number of 1 notes is: " + notesOf1);
    }
}

File: Basic Coding Problems/01 Fundamentals/018 Write a program to count the total number of notes in a given amount/C++/count_notes.cpp
#include <iostream>

using namespace std;

int main() {
    int amount;
    int notes_of_500, notes_of_200, notes_of_100, notes_of_50, notes_of_20, notes_of_10, notes_of_5, notes_of_2, notes_of_1;

    cout << "Enter the amount: ";
    cin >> amount;

    notes_of_500 = amount / 500;
    amount %= 500;

    notes_of_200 = amount / 200;
    amount %= 200;

    notes_of_100 = amount / 100;
    amount %= 100;

    notes_of_50 = amount / 50;
    amount %= 50;

    notes_of_20 = amount / 20;
    amount %= 20;

    notes_of_10 = amount / 10;
    amount %= 10;

    notes_of_5 = amount / 5;
    amount %= 5;

    notes_of_2 = amount / 2;
    amount %= 2;

    notes_of_1 = amount;

    cout << "The number of 500 notes is: " << notes_of_500 << endl;
    cout << "The number of 200 notes is: " << notes_of_200 << endl;
    cout << "The number of 100 notes is: " << notes_of_100 << endl;
    cout << "The number of 50 notes is: " << notes_of_50 << endl;
    cout << "The number of 20 notes is: " << notes_of_20 << endl;
    cout << "The number of 10 notes is: " << notes_of_10 << endl;
    cout << "The number of 5 notes is: " << notes_of_5 << endl;
    cout << "The number of 2 notes is: " << notes_of_2 << endl;
    cout << "The number of 1 notes is: " << notes_of_1 << endl;

    return 0;
}